app/views/search/index.html.erb

```diff
@@ -5,57 +5,65 @@
 <div class="row">
   <div class="col-lg-6 col-md-9 col-12 mx-auto">
 
-    <%= render('partials/search_bar', :locals => {:path => search_path}) %>
+    <%= render('partials/search_bar', :locals => { :path => search_path }) %>
 
     <div class="row">
       <div class="col">
         <div class="list-group" id="exact-results">
-          <%if @exact_results.present? %>
-            <% @exact_results.each do |result|%>
-
-              <a href="<%=url_for(result)%>" class="list-group-item list-group-item-action search-item">
-                <div class='row'>
-                  <div class='col-2 d-flex flex-column'>
-                    <%= image_tag(result.image_or_placeholder, class: "picture-rounded md") %>
-                  </div>
-                  <div class="col d-flex flex-column my-2">
-                    <h5 class="search-title"><%= result.to_string %></h5>
-                    <div>
-                      <% result.displayed_tags.each do |tag| %>
-                        <span class="badge bg-primary"><%= tag %></span>
-                      <% end %>
+          <% if @exact_results.present? %>
+            <% @exact_results.each do |result| %>
+              <% unless result.class == Person and result.role == "student" and not user_signed_in? %>
+                <a href="<%= url_for(result) %>" class="list-group-item list-group-item-action search-item">
+                  <div class='row'>
+                    <div class='col-2 d-flex flex-column'>
+                      <%= image_tag(result.image_or_placeholder, class: "picture-rounded md") %>
+                    </div>
+                    <div class="col d-flex flex-column my-2">
+                      <h5 class="search-title"><%= result.to_string %></h5>
+                      <div>
+                        <% result.displayed_tags.each do |tag| %>
+                          <% def prohibited_keys
+                               ["Telegram"]
```

Comment:
This is where you can add attributes, that you don't want not signed in users to see.